File: src/main/java/com/duxtinto/intellij/plugin/github/codereviews/ui/toolwindow/pullrequestlist/columns/ColumnInfoFactory.java
package com.duxtinto.intellij.plugin.github.codereviews.ui.toolwindow.pullrequestlist.columns;

import com.duxtinto.intellij.plugin.github.codereviews.domain.pullrequests.PullRequestEntity;
import com.intellij.util.ui.ColumnInfo;

import javax.inject.Inject;

public class ColumnInfoFactory {
    @Inject
    public ColumnInfoFactory() {
    }

    public enum ColumnIndexes {
        NUMBER,
        ISSUE,
        STATE,
        TITLE,
    }

    public ColumnInfo<PullRequestEntity, ?>[] createDefaultColumns() {
        return new ColumnInfo[]{
                new NumberColumnInfo(),
                new IssuesColumnInfo(),
                new StateColumnInfo(),
                new TitleColumnInfo(),
        };
    }
}

File: src/main/kotlin/com/duxtinto/intellij/plugin/github/codereviews/presentation/toolwindow/pullrequestlist/columns/ColumnInfoFactory.kt
package com.duxtinto.intellij.plugin.github.codereviews.presentation.toolwindow.pullrequestlist.columns

import com.duxtinto.intellij.plugin.github.codereviews.domain.pullrequests.PullRequestEntity
import com.intellij.util.ui.ColumnInfo

import javax.inject.Inject

class ColumnInfoFactory
    @Inject constructor() {

    enum class ColumnIndexes {
        NUMBER,
        ISSUE,
        STATE,
        TITLE
    }

    fun createDefaultColumns(): Array<ColumnInfo<PullRequestEntity, *>> {
        return arrayOf(NumberColumnInfo(), IssuesColumnInfo(), StateColumnInfo(), TitleColumnInfo())
    }
}
